src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploaddocuments/UploadDocumentWorkAllocationHandlerTest.java

```diff
@@ -0,0 +1,141 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.uploaddocuments;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType.ABOUT_TO_SUBMIT;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Optional;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.CaseDetails;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsDocument;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsDocumentDetails;
+import uk.gov.hmcts.reform.sscs.ccd.domain.State;
+import uk.gov.hmcts.reform.sscs.util.AddedDocumentsUtil;
+
+@RunWith(JUnitParamsRunner.class)
+public class UploadDocumentWorkAllocationHandlerTest {
+
+    private static final String USER_AUTHORISATION = "Bearer token";
+
+    private UploadDocumentWorkAllocationHandler handler = new UploadDocumentWorkAllocationHandler(
+            new AddedDocumentsUtil(true), true);
+
+    @Before
+    public void setUp() {
+    }
+
+    @Test
+    public void givenAnInvalidEvent_thenReturnFalse() {
+        assertFalse(handler.canHandle(ABOUT_TO_SUBMIT, createCallBack(EventType.VALID_APPEAL_CREATED, null, null)));
+    }
+
+    @Test
+    @Parameters({"ABOUT_TO_START", "MID_EVENT", "SUBMITTED"})
+    public void givenANonCallbackType_thenReturnFalse(CallbackType callbackType) {
+        assertFalse(handler.canHandle(callbackType, createCallBack(null, null)));
+    }
+
+    @Test
+    public void givenNewDocumentAddedNoPreviousCaseData_thenSetScannedDocumentTypes() {
+        SscsCaseData sscsCaseData = SscsCaseData.builder()
+                .sscsDocument(Arrays.asList(
+                        SscsDocument.builder().id("111-111").value(SscsDocumentDetails.builder().documentType("confidentialityRequest").build()).build()
```

Comment:
Minor comment - avoid nested builders if possible, separate them out for easier readability / amendment.